api/client/build.go

```diff
@@ -390,36 +278,209 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	return err
 }
 
-var (
-	dockerfileFromLinePattern        = regexp.MustCompile(`(?i)^FROM[ \f\r\t\v]+(?P<image>[^ \f\r\t\v\n#]+)`)
-	dockerfileLineContinutionPattern = regexp.MustCompile(`\\[ \t]*$`)
-)
+func getDockerfileRelPath(dockerfileName *string, baseDir string) (err error) {
```

Comment:
I'm redoing most of these so the pointer type wont be an issue. I'm also going through and adding comments to each of the added functions.